Your pull request does not meet our code style rules. Pylint summary:
```
************* Module ase_interface
testsuite/python/ase_interface.py:20:0: W0611: Unused unittest_decorators imported as utx (unused-import)
```

You can generate these warnings with `maintainer/CI/fix_style.sh`. This is the same command that I have executed to generate the log above.

--- a/src/python/espressomd/CMakeLists.txt
+++ b/src/python/espressomd/CMakeLists.txt
@@ -30,6 +30,7 @@ set(cython_AUX "${cython_AUX}" CACHE INTERNAL "cython_AUX")
 
 add_subdirectory(io)
 add_subdirectory(detail)
+add_subdirectory(plugins)
 
 list(REMOVE_DUPLICATES cython_SRC)
 

--- a/src/python/espressomd/plugins/CMakeLists.txt
+++ b/src/python/espressomd/plugins/CMakeLists.txt
@@ -1,0 +1,23 @@
+#
+# Copyright (C) 2024 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+configure_file(ase.py ase.py COPYONLY)
+set(cython_AUX ${cython_AUX}
+               "${CMAKE_SOURCE_DIR}/src/python/espressomd/io/__init__.py"
+    CACHE INTERNAL "cython_AUX" FORCE)

--- a/src/python/espressomd/plugins/ase.py
+++ b/src/python/espressomd/plugins/ase.py
@@ -1,0 +1,69 @@
+#
+# Copyright (C) 2024 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import dataclasses
+import typing
+import ase
+from ase.calculators.singlepoint import SinglePointCalculator
+import numpy as np
+if typing.TYPE_CHECKING:
+    from espressomd.system import System
+
+
+@dataclasses.dataclass
+class ASEInterface:
+    """
+    ASE interface for ESPResSo.
+    """
+
+    type_mapping: dict
+    """
+    Mapping of ESPResSo particle types to ASE symbols. E.g. ``{0: "H", 1: "O"}``.
+    """
+    _system: typing.Union["System", None] = None
+
+    def register_system(self, system):
+        """Register the system."""
+        self._system = system
+
+    def __getstate__(self):
+        return {"type_mapping": self.type_mapping}
+
+    def get(self) -> ase.Atoms:
+        """Export the ESPResSo system particle data to an ASE atoms object."""
+        particles = self._system.part.all()
+        positions = np.copy(particles.pos)
+        types = np.copy(particles.type)
+        forces = np.copy(particles.f)
+        unknown_types = set(types) - set(self.type_mapping)
+        if unknown_types:
+            raise RuntimeError(
+                f"Particle types '{unknown_types}' haven't been registered in the ASE type map"  # nopep8
+            )
+        if any(p.is_virtual() for p in particles):
+            raise RuntimeError("ASE doesn't support virtual sites")
+
+        atoms = ase.Atoms(
+            positions=positions,
+            symbols=[self.type_mapping[t] for t in types],
+            pbc=np.copy(self._system.periodicity),
+            cell=np.copy(self._system.box_l),
+        )
+        atoms.calc = SinglePointCalculator(atoms, forces=forces)
+        return atoms

--- a/src/python/espressomd/system.py
+++ b/src/python/espressomd/system.py
@@ -200,6 +200,7 @@ class System(ScriptInterfaceHelper):
         self.non_bonded_inter = interactions.NonBondedInteractions()
         self.part = particle_data.ParticleList()
         self.thermostat = thermostat.Thermostat()
+        self._ase_interface = None
 
         # lock class
         self.call_method("lock_system_creation")
@@ -238,11 +239,17 @@ class System(ScriptInterfaceHelper):
         odict["_system_handle"] = self.call_method("get_system_handle")
         for property_name in checkpointable_properties:
             odict[property_name] = System.__getattribute__(self, property_name)
+        if self._ase_interface is not None:
+            odict["_ase_interface"] = self._ase_interface.__getstate__()
         return odict
 
     def __setstate__(self, params):
         # note: this class is initialized twice by pickle
         self.call_method("set_system_handle", obj=params.pop("_system_handle"))
+        # initialize Python-only members
+        if "_ase_interface" in params:
+            from espressomd.plugins.ase import ASEInterface
+            self.ase = ASEInterface(**params.pop("_ase_interface"))
         for property_name in params.keys():
             System.__setattr__(self, property_name, params[property_name])
         # note: several members can only be instantiated once
@@ -337,6 +344,15 @@ class System(ScriptInterfaceHelper):
             if lb is not None:
                 lb.call_method("activate")
                 self._lb = lb
+
+    @property
+    def ase(self):
+        return self._ase_interface
+
+    @ase.setter
+    def ase(self, ase):
+        ase.register_system(self)
+        self._ase_interface = ase
 
     @property
     def ekcontainer(self):

--- a/testsuite/python/CMakeLists.txt
+++ b/testsuite/python/CMakeLists.txt
@@ -202,6 +202,7 @@ checkpoint_test(MODES dp3m_cpu__therm_langevin__int_nvt)
 checkpoint_test(MODES therm_dpd__int_nvt)
 checkpoint_test(MODES scafacos__therm_bd__int_bd)
 checkpoint_test(MODES therm_sdm__int_sdm)
+checkpoint_test(MODES lj__ase SUFFIX 1_core MAX_NUM_PROC 1)
 
 python_test(FILE bond_breakage.py MAX_NUM_PROC 4)
 python_test(FILE cell_system.py MAX_NUM_PROC 4)

--- a/testsuite/python/ase_interface.py
+++ b/testsuite/python/ase_interface.py
@@ -1,0 +1,69 @@
+#
+# Copyright (C) 2024 The ESPResSo project
+#
+# This file is part of ESPResSo.
+#
+# ESPResSo is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ESPResSo is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+import unittest as ut
+import unittest_decorators as utx
+import espressomd
+import espressomd.plugins.ase
+import numpy as np
+import ase
+
+
+class ASEInterfaceTest(ut.TestCase):
+
+    system = espressomd.System(
+        box_l=[10., 10., 10.], periodicity=[True, True, False])
+
+    def setUp(self):
+        self.system.part.add(pos=[0., 0., 0.], f=[1., -1., 0.], type=0)
+        self.system.part.add(pos=[0., 0., 1.], f=[0., 12., 0.], type=1)
+        self.system.ase = espressomd.plugins.ase.ASEInterface(
+            type_mapping={0: "H", 1: "O"},
+        )
+
+    def tearDown(self):
+        self.system.part.clear()
+
+    def test_ase_get(self):
+        """Test the ``ASEInterface.get()`` method."""
+        # Create a simple ASE atoms object
+        atoms = self.system.ase.get()
+        self.assertIsInstance(atoms, ase.Atoms)
+        self.assertEqual(set(atoms.get_chemical_symbols()), {"H", "O"})
+        np.testing.assert_equal(atoms.pbc, np.copy(self.system.periodicity))
+        np.testing.assert_allclose(atoms.cell, np.diag(self.system.box_l))
+        np.testing.assert_allclose(atoms.get_positions(),
+                                   [[0., 0., 0.], [0., 0., 1.]])
+        np.testing.assert_allclose(atoms.get_forces(),
+                                   [[1., -1., 0.], [0., 12., 0.]])
+
+    @utx.skipIfMissingFeatures("VIRTUAL_SITES_RELATIVE")
+    def test_exceptions(self):
+        p = self.system.part.add(pos=[0., 0., 0.], type=10)
+        with self.assertRaisesRegex(RuntimeError, r"Particle types '\{10\}' haven't been registered"):
+            self.system.ase.get()
+        p.type = 1
+        vs = self.system.part.add(pos=[0., 0., 0.], type=1)
+        vs.vs_relative = (p.id, 0.01, (1., 0., 0., 0.))
+        with self.assertRaisesRegex(RuntimeError, "ASE doesn't support virtual sites"):
+            self.system.ase.get()
+
+
+if __name__ == "__main__":
+    ut.main()

--- a/testsuite/python/save_checkpoint.py
+++ b/testsuite/python/save_checkpoint.py
@@ -20,8 +20,10 @@
 import unittest as ut
 import unittest_generator as utg
 import numpy as np
+import contextlib
 import pathlib
 import tempfile
+import sys
 
 import espressomd
 import espressomd.checkpointing
@@ -40,6 +42,9 @@ import espressomd.shapes
 import espressomd.constraints
 import espressomd.bond_breakage
 import espressomd.reaction_methods
+
+with contextlib.suppress(ImportError):
+    import espressomd.plugins.ase
 
 config = utg.TestGenerator()
 modes = config.get_modes()
@@ -71,6 +76,8 @@ if 'INT.NPT' not in modes and 'LB.GPU' not in modes:
         initial_pos_offset=0.1, time_0=0.2, shear_velocity=1.2)
     system.lees_edwards.set_boundary_conditions(
         shear_direction="x", shear_plane_normal="y", protocol=protocol)
+
+has_ase = "ASE" in modes
 
 lbf_class = None
 lb_lattice = None
@@ -161,6 +168,11 @@ if espressomd.has_features('P3M') and ('P3M' in modes or 'ELC' in modes):
         system.electrostatics.solver = p3m
         p3m.charge_neutrality_tolerance = 5e-12
 
+if "ase" in sys.modules:
+    system.ase = espressomd.plugins.ase.ASEInterface(
+        type_mapping={0: "H", 1: "O", 10: "Cl"},
+    )
+
 # accumulators
 obs = espressomd.observables.ParticlePositions(ids=[0, 1])
 acc_mean_variance = espressomd.accumulators.MeanVarianceCalculator(obs=obs)
@@ -240,7 +252,7 @@ if 'LB' not in modes:
             approximation_method='ft', viscosity=0.5, radii={0: 1.5},
             pair_mobility=False, self_mobility=True)
 
-if espressomd.has_features(['VIRTUAL_SITES', 'VIRTUAL_SITES_RELATIVE']):
+if espressomd.has_features(['VIRTUAL_SITES_RELATIVE']) and not has_ase:
     p2.vs_auto_relate_to(p1)
 
 # non-bonded interactions
@@ -451,6 +463,7 @@ if espressomd.has_features('THERMOSTAT_PER_PARTICLE'):
 if espressomd.has_features(["ENGINE"]):
     p3.swimming = {"f_swim": 0.03}
 if espressomd.has_features(["ENGINE", "VIRTUAL_SITES_RELATIVE"]) and lbf_class:
+    assert not has_ase
     p4.swimming = {"v_swim": 0.02, "is_engine_force_on_fluid": True}
 if espressomd.has_features('LB_ELECTROHYDRODYNAMICS') and lbf_class:
     p8.mu_E = [-0.1, 0.2, -0.3]

--- a/testsuite/python/test_checkpoint.py
+++ b/testsuite/python/test_checkpoint.py
@@ -42,6 +42,10 @@ with contextlib.suppress(ImportError):
     import espressomd.io.vtk
 
 with contextlib.suppress(ImportError):
+    import ase
+    import espressomd.plugins.ase
+
+with contextlib.suppress(ImportError):
     import h5py  # h5py has to be imported *after* espressomd (MPI)
 
 config = utg.TestGenerator()
@@ -53,6 +57,7 @@ has_p3m_mode = 'P3M.CPU' in modes or 'P3M.GPU' in modes and is_gpu_available
 has_thermalized_bonds = 'THERM.LB' in modes or 'THERM.LANGEVIN' in modes
 has_drude = (espressomd.has_features(['ELECTROSTATICS' and 'MASS', 'ROTATION'])
              and has_thermalized_bonds)
+has_ase = 'ASE' in modes
 
 
 class CheckpointTest(ut.TestCase):
@@ -448,13 +453,13 @@ class CheckpointTest(ut.TestCase):
                 p3.swimming,
                 {"f_swim": 0.03, "is_engine_force_on_fluid": False})
             if espressomd.has_features(
-                    'VIRTUAL_SITES_RELATIVE') and has_lb_mode:
+                    'VIRTUAL_SITES_RELATIVE') and has_lb_mode and not has_ase:
                 self.assertEqual(
                     p4.swimming,
                     {"f_swim": 0., "is_engine_force_on_fluid": True})
         if espressomd.has_features('LB_ELECTROHYDRODYNAMICS') and has_lb_mode:
             np.testing.assert_allclose(np.copy(p8.mu_E), [-0.1, 0.2, -0.3])
-        if espressomd.has_features('VIRTUAL_SITES_RELATIVE'):
+        if espressomd.has_features('VIRTUAL_SITES_RELATIVE') and not has_ase:
             from scipy.spatial.transform import Rotation as R
             q_ind = ([1, 2, 3, 0],)  # convert from scalar-first to scalar-last
             vs_id, vs_dist, vs_quat = p2.vs_relative
@@ -739,6 +744,7 @@ class CheckpointTest(ut.TestCase):
         self.assertEqual(dh.drude_id_list, [5])
 
     @utx.skipIfMissingFeatures(['VIRTUAL_SITES', 'VIRTUAL_SITES_RELATIVE'])
+    @ut.skipIf("ASE" in modes, "virtual sites not allowed by ASE")
     def test_virtual_sites(self):
         Propagation = espressomd.propagation.Propagation
         p_real = system.part.by_id(0)
@@ -1022,6 +1028,22 @@ class CheckpointTest(ut.TestCase):
         p1.remove()
         p2.remove()
         system.non_bonded_inter[2, 6].reset()
+
+    @ut.skipIf("ase" not in sys.modules, "missing module 'ase'")
+    @ut.skipIf("ASE" not in modes, "missing combination")
+    def test_ase_plugin(self):
+        atoms = system.ase.get()
+        self.assertIsNotNone(atoms)
+        self.assertIsInstance(atoms, ase.Atoms)
+        self.assertEqual(set(atoms.get_chemical_symbols()), {"H", "O"})
+        np.testing.assert_equal(atoms.pbc, np.copy(system.periodicity))
+        np.testing.assert_allclose(atoms.cell, np.diag(system.box_l))
+        np.testing.assert_allclose(
+            atoms.get_positions(),
+            np.copy(system.part.select(lambda p: p.type in [0, 1]).pos))
+        np.testing.assert_allclose(
+            atoms.get_forces(),
+            np.copy(system.part.select(lambda p: p.type in [0, 1]).f))
 
 
 if __name__ == '__main__':

